spark/src/main/java/org/apache/iceberg/spark/source/PositionDeleteFileReader.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark.source;
+
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.avro.Avro;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.io.InputFile;
+import org.apache.iceberg.orc.ORC;
+import org.apache.iceberg.parquet.Parquet;
+import org.apache.iceberg.spark.data.SparkAvroReader;
+import org.apache.iceberg.spark.data.SparkOrcReader;
+import org.apache.iceberg.spark.data.SparkParquetReaders;
+import org.apache.iceberg.types.Types;
+import org.apache.spark.sql.catalyst.InternalRow;
+
+import static org.apache.iceberg.types.Types.NestedField.required;
+
+class PositionDeleteFileReader {
```

Comment:
Why is this in the Spark package/module?

Reply:
I explained here https://github.com/apache/iceberg/pull/971#discussion_r454396662.  This should move to core and use GenericParquetReader.

Reply:
I'm not quite sure I understand what you mean by that comment. Are you saying that we no longer need the reader because we just need to use generics? If so, I think I agree because this is quite simple. Maybe we should just add the writer and not a reader.

Reply:
I meant to move `PositionDeleteFileReader` to `iceberg-data` so that it could be used by any engine.  Similarly, the writer can be implemented in `iceberg-data` as well.